tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/mapper/CurriculumMembershipMapper.java

```diff
@@ -222,29 +214,33 @@ private CurriculumMembership programmeMembershipDtoToCurriculumMembership(
       ProgrammeMembershipDTO programmeMembershipDto) {
     CurriculumMembership result = new CurriculumMembership();
 
-    result.setId(programmeMembershipDto.getId());
+    //TODO: the following should be removed in future once e.g. the tcs-persistence repository queries are refactored
+    // to allow CurriculumMembership to not duplicate ProgrammeMembership data
     result.setProgrammeMembershipType(programmeMembershipDto.getProgrammeMembershipType());
-    result.setRotation(rotationDtoToRotation(programmeMembershipDto.getRotation()));
     result.setProgrammeStartDate(programmeMembershipDto.getProgrammeStartDate());
     result.setProgrammeEndDate(programmeMembershipDto.getProgrammeEndDate());
-    result.setLeavingDestination(programmeMembershipDto.getLeavingDestination());
     result.setLeavingReason(programmeMembershipDto.getLeavingReason());
-    result.setTrainingPathway(programmeMembershipDto.getTrainingPathway());
+    result.setLeavingDestination(programmeMembershipDto.getLeavingDestination());
     if (programmeMembershipDto.getProgrammeId() != null) {
       Programme programme = new Programme();
       programme.setId(programmeMembershipDto.getProgrammeId());
       programme.setProgrammeName(programmeMembershipDto.getProgrammeName());
+      programme.setOwner(programmeMembershipDto.getProgrammeOwner());
       programme.setProgrammeNumber(programmeMembershipDto.getProgrammeNumber());
       result.setProgramme(programme);
+      result.setRotation(rotationDtoToRotation(programmeMembershipDto.getRotation()));
     }
     result.setTrainingNumber(
-        trainingNumberDtoToTrainingNumber(programmeMembershipDto.getTrainingNumber()));
+        trainingNumberDTOToTrainingNumber(programmeMembershipDto.getTrainingNumber()));
```

Comment:
Can this change be reverted?

Reply:
Sure. As you can see, the function was deleted and then later reinstated with the same name as the equivalent function in ProgrammeMembershipMapper. 🤦